### PR TITLE
Unlock Loot items directly from dw server

### PR DESCRIPTION
**Pull request by hosseinpourziyaie**
shitty way for unlocking items. it can be used to generate dummie for tracing item flow and pinpoint loot handler functions using debugger

### CODE REVIEW

File: src/client/game/demonware/services/bdMarketplace.cpp [hunk 1]
```diff
@@ -78,8 +78,32 @@ namespace demonware
...
+		LootID_ranges.insert({ 110005, 110162 }); // mp\loot\iw7_cosmetic_emotes_loot_master.csv
+		
+		for (auto const& pair : LootID_ranges)
+		{
+			for (size_t loot_id = pair.first; loot_id <= pair.second; loot_id++)
+			{
+				auto loot_item = new bdMarketplaceInventory{};
+				loot_item->m_userID = 0;
+				loot_item->m_accountType = "iw_pc_steam";
+				loot_item->m_itemId = loot_id;
```
Comment (Joelrau) [inline src/client/game/demonware/services/bdMarketplace.cpp:94]: use static_cast

**Comment by Joelrau** [comment]
so is this supposed to unlock all items?

**Comment by hosseinpourziyaie** [comment]
> so is this supposed to unlock all items?

check in code. a loop spins and adds list of loot id range.
you can add any loot table range but given range shouldnt have any gaps(unused placeholder) or game simply will freeze 
however i should mention that current shitty dw emulator in-use has poor handling about oversized reply buffers so you shouldnt really add too much. 

Also this is not really intended to unlock stuff. it has experimental purpose as i mentioned before.

**Comment by Joelrau** [comment]
since this is only experimental currently, then i won't merge it.

**Pull request closed without merging**